### ISSUE TITLE
Missing semantic HTML attributes on StreamField form widgets

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10463]
### Issue Summary

In the HTML output of StreamField form widgets, we’re missing a lot of important HTML attributes compared to model-level form fields:

- `required` for required fields
- `aria-invalid="true"` for fields with errors
- `aria-describedby` for help text
- `aria-describedby` for error messages

All `input type="text"` in StreamField seem to only ever render as:

```html
<input type="text" name="backstory-0-value-heading_text" id="backstory-0-value-heading_text">
```

Similarly a required `textarea` with an error message is:

```html
<textarea name="backstory-2-value-text" cols="40" rows="1" data-controller="w-autosize" class="w-field__autosize" id="backstory-2-value-text" style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 71px;"></textarea>
```

For comparison, here’s the markup for our page `title` field (excluding Stimulus attributes):

```html
<input type="text" name="title" placeholder="Page title*" maxlength="255" aria-invalid="true" aria-describedby="panel-child-content-child-title-helptext panel-child-content-child-title-errors" required="" id="id_title">
```

### Steps to Reproduce

1. Go to the page editor for a bakerydemo RecipePage
2. Add an empty heading block & blockquote
3. Save draft

**Comment by thibaudcolas** [comment]
Guess what, I forgot I had already reported this :) See #10300
